Added rehook-segment documentation (#5219)
segmentio/segment-docs: src/connections/destinations/catalog/rehook/index.md

---
title: Rehook Destination
rewrite: true
id: 64c02312ff0ce798cc8d1a7e
---

> 

Reviewer: ```suggestion
{% include content/plan-grid.md name="actions" %}
```

Author: done

[Rehook](https://rehook.ai/){:target="_blank"} is a powerful and dedicated user-incentivization solution that enables businesses to reward and engage users without any dependency on tech teams. With an elegant and easy-to-use interface, Rehook is designed to help you run user-promotion campaigns that are flexible, customizable and scalable.

Rehook maintains this destination; for any issues with the destination, [contact the Rehook Support team](mailto:services@rehook.ai).

## Getting Started

1. From the Destinations catalog page in the Segment App, click **Add Destination**.
2. Search for **Rehook** in the Destinations Catalog, and select the **Rehook** destination.
3. Choose which Source should send data to the Rehook destination.
4. Go to the [Rehook Dashboard](https://dashboard.rehook.ai/){:target="_blank"} and navigate to **Settings > API Keys & Secret Key**, and copy it.
5. Enter the **API Key & Secret Key** in the Rehook destination settings in Segment.

## Supported methods

> Rehook supports the following methods, as specified in the [Segment Spec](/docs/connections/spec).

Reviewer: ```suggestion
Rehook's destination is designed to support the following methods, as specified in the [Segment Spec](/docs/connections/spec). Note that because this is an Actions Destination, you will also have the capability to map other Segment methods if neccessary.
```

Author: done

> * The userId field is required. Rehook drops identify calls without a userId.
> * If a call is made with anonymousID, Rehook drops the identify call.
> * If referral_code matches with another userId, Rehook drops the identify call.

### Track

Take a look at the [Track method documentation](/docs/connections/spec/track) to learn about what it does. An example call looks like:

#### Example 1:
```js
analytics.track('Product Viewed', {
  userId: "userId12345",
  product_id: '507f1f779439011',
  name: 'Monopoly: 3rd Edition',
  price: 18.99,
  url: 'https://www.example.com/product/path',
  image_url: 'https://www.example.com/product/path.jpg'
});
```

#### Example 2:
```js
analytics.track('signup', {
  userId: "userId12345",
  referral_code: 'ERTYUS'
});
```

Segment sends `Track` calls to Rehook as a Custom Event. When you call the track, Segment sends the event to Rehook with the event name and all properties that you specified.

> info "How Rehook handles incoming userId and referral_code in track call"
> * The userId field is required. Rehook drops track calls without a userId.
> * If a call is made with anonymousId, Rehook drops the track call.

> > * The referral_code field is required, if event name is set as a conversion event on Rehook. 

Reviewer: ```suggestion
> * The referral_code field is required, if event name is set as a conversion event on Rehook. 

{% include components/actions-fields.html %}
```

Author: done
